ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala

```diff
@@ -465,59 +465,32 @@ object JsonProtocol extends JsonProtocolLow {
     ErrorInfoDetail
   )
 
-  implicit val RetryInfoDetailFormat: RootJsonFormat[RetryInfoDetail] =
-    new RootJsonFormat[RetryInfoDetail] {
-      override def write(obj: RetryInfoDetail): JsValue = JsObject(
-        "duration" -> JsNumber(obj.duration.toNanos)
+  // Encapsulating the way how duration is serialized to be able to reimport
+  // where necessary.
+  object RetryInfoDetailFormats {
+    import scala.concurrent.duration.Duration
+    implicit val durationFormat: JsonFormat[Duration] =
```

Comment:
I would need to duplicate the type definition for that in that case. I can do that but so far wasn't keen on doing that because of the duplication and more indirection etc.

Reply:
Done with [d317efe](https://github.com/digital-asset/daml/pull/13770/commits/d317efe1d95e35fe347263f568689532bd95070c).